tests/test_search.py

```diff
@@ -2073,6 +2093,96 @@ def test_profile(client):
 
 @pytest.mark.redismod
 @pytest.mark.onlynoncluster
+@skip_if_redis_enterprise()
+@skip_if_server_version_lt("7.9.0")
+def test_profile_with_coordinator(client):
+    client.ft().create_index((TextField("t"),))
+    client.ft().client.hset("1", "t", "hello")
+    client.ft().client.hset("2", "t", "world")
+
+    # check using Query
+    q = Query("hello|world").no_content()
+    if is_resp2_connection(client):
+        res, det = client.ft().profile(q)
+        det = det.info
+        assert det[0] == "Shards"
```

Comment:
Arguably, we shouldn't test FT.PROFILE reply structure, since we are testing the client here, not the server. I'd suggest trimming most of the checks and replace them with something like 
```python
assert typeof(det) is dict
assert len(det) >= X
```